### PR TITLE
Add helm index for v1.10.0 and v1.11.0

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add helm index for v1.10.0 and v1.11.0

these two new version can't be searched by helm, as you see:

```shell
$ helm search repo karmada-charts -l
NAME                            CHART VERSION   APP VERSION     DESCRIPTION                      
karmada-charts/karmada          v1.9.0          latest          A Helm chart for karmada         
karmada-charts/karmada          v1.8.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.7.1          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.7.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.6.2          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.6.1          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.6.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.5.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.4.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.3.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.2.2          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.2.1          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada          v1.2.0          v1.1.0          A Helm chart for karmada         
karmada-charts/karmada-operator v1.8.0          v1.1.0          A Helm chart for karmada-operator
```

**Which issue(s) this PR fixes**:

Fixes part of #5484

**Special notes for your reviewer**:

1. generate methods: https://github.com/karmada-io/karmada/pull/4051#discussion_r1324310397

2. test report on my local branch:

```shell
$ helm repo add my-branch-karmada https://raw.githubusercontent.com/chaosi-zju/karmada/chart-index/charts
"my-branch-karmada" has been added to your repositories

$ helm repo list        
NAME                            URL                                                                    
...
my-branch-karmada               https://raw.githubusercontent.com/chaosi-zju/karmada/chart-index/charts

$ helm search repo my-branch-karmada -l
NAME                                    CHART VERSION   APP VERSION     DESCRIPTION                      
my-branch-karmada/karmada               v1.11.0         v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.10.0         v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.9.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.8.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.7.1          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.7.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.6.2          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.6.1          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.6.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.5.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.4.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.3.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.2.2          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.2.1          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada               v1.2.0          v1.1.0          A Helm chart for karmada         
my-branch-karmada/karmada-operator      v1.11.0         v1.1.0          A Helm chart for karmada-operator
my-branch-karmada/karmada-operator      v1.10.0         v1.1.0          A Helm chart for karmada-operator
my-branch-karmada/karmada-operator      v1.9.0          v1.1.0          A Helm chart for karmada-operator
my-branch-karmada/karmada-operator      v1.8.0          v1.1.0          A Helm chart for karmada-operator
```

**Does this PR introduce a user-facing change?**:

```release-note
`Helm chart`: Added helm index for v1.10.0 and v1.11.0 release.
```

### CODE REVIEW

File: charts/index.yaml [hunk 1]
```diff
@@ -2,7 +2,51 @@ apiVersion: v1
...
+    description: A Helm chart for karmada
+    digest: d4d9bbbda5ac03a4ba9349b17973b3df9047849398735dcb36cec4ec39140046
+    kubeVersion: '>= 1.16.0-0'
+    maintainers:
+    - email: jrkeen@hotmail.com
+      name: jrkeen
+    - email: jackson.cloudnative@gmail.com
+      name: pidb
+    - email: shentiecheng@huawei.com
+      name: Poor12
```
Comment (RainbowMango) [inline charts/index.yaml:20]: I think you should put you and @a7i on the maintainer's list.

File: charts/index.yaml [hunk 2]
```diff
@@ -2,7 +2,51 @@ apiVersion: v1
...
+    name: karmada
+    type: application
+    urls:
+    - https://github.com/karmada-io/karmada/releases/download/v1.11.0/karmada-chart-v1.11.0.tgz
+    version: v1.11.0
```
Comment (zhzhuang-zju) [inline charts/index.yaml:25]: Whether we need to add the Helm index for a patch release like `v1.11.1`.

**Comment by a7i** [comment]
this is pretty easy to automate, if you're still up for it. looks what we've done on the Descheduler side: https://github.com/kubernetes-sigs/descheduler/tree/gh-pages

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

It would be great if we could automate it.

**Pull request merged**